h/views/account_signup.py

```diff
@@ -237,13 +243,13 @@ def post(self):
             identities=[
                 {
                     "provider": self.settings.provider,
-                    "provider_unique_id": provider_unique_id,
+                    "provider_unique_id": idinfo.sub,
                 }
             ],
         )
 
         return HTTPFound(
-            self.request.route_url("activity.user_search", username=user.username),
+            redirect_url(self.request, user, idinfo.next_url),
```

Comment:
This redirects to the URL from the `next` query param after sucessfully  signing up using a social login option. (At this point in the flow the `next` URL is being retrieved from the `idinfo` token query param, it's no longer in a `next` query param.)